proposals/lip-0052.md

```diff
@@ -362,712 +377,1926 @@ The module provides the following commands to modify the NFT store.
-        if chainID == CHAIN_ID_ALIAS_NATIVE:
-            existingIndex[collection] = max{existingIndex[collection], index}
-
-    # Check that all existing indices are smaller than the next available index for the collection
-    for collection in existingIndex:
-        if (collectionStore(collection) does not exist
-            or collectionStore(collection).nextAvailableIndex <= existingIndex[collection]):
-            return False
-
-    # Check that the largest existing collection is smaller than the recorded available collection
-    maxCollection = max key of the existingIndex dictionary, -1 if existingIndex is empty
-    # Recall that nextAvailableCollection is used as the value of the available collection substore
-    if nextAvailableCollection <= maxCollection:
-        return False
-
-    return True
+def execute(
+    ccu: Transaction,
+    ccm: CCM
+) -> None:
+
+    nftID = ccm.params.nftID
+    senderAddress = ccm.params.senderAddress
+    recipientAddress = ccm.params.recipientAddress
+    receivedAttributes = ccm.params.attributesArray
+    data = ccm.params.data
+    receivingChainID = ccm.receivingChainID
+    sendingChainID = ccm.sendingChainID
+
+    if getChainID(nftID) == OWN_CHAIN_ID: # execution on the native chain
+        storedAttributes = NFTStore[nftID].attributesArray
+        if ccm.status == CCM_STATUS_CODE_OK:
+            NFTStore[nftID].owner = recipientAddress
+            NFTStore[nftID].attributesArray = getNewAttributes(nftID, storedAttributes, receivedAttributes)
+            createUserEntry(recipientAddress, nftID)
+            delete entry escrowEntry(sendingChainID, nftID) from the escrow substore
+        else: # return the NFT to the sender
+            recipientAddress = senderAddress
+            NFTStore[nftID].owner = recipientAddress
+            createUserEntry(recipientAddress, nftID)
+            delete entry escrowEntry(sendingChainID, nftID) from the escrow substore
+    else: # execution on the foreign chain
+        if isNFTSupported(nftID) == False:
+            emitPersistentEvent(
+                module = MODULE_NAME_NFT,
+                name = EVENT_NAME_CCM_TRANSFER,
+                data = {
+                    "senderAddress": senderAddress,
+                    "recipientAddress": recipientAddress,
+                    "nftID": nftID,
+                    "result": RESULT_NFT_NOT_SUPPORTED
+                },
+                topics = [senderAddress, recipientAddress]
+            )
+            raise Exception("Non-supported NFT")
+        if ccm.status == CCM_STATUS_CODE_OK:
+            Fee.payFee(FEE_CREATE_NFT)
+            createNFTEntry(recipientAddress, nftID, receivedAttributes)
+            createUserEntry(recipientAddress, nftID)
+        else: # return the NFT to the sender
+            recipientAddress = senderAddress
+            createNFTEntry(recipientAddress, nftID, receivedAttributes)
+            createUserEntry(senderAddress, nftID)
```

Comment:
```suggestion
            createUserEntry(recipientAddress, nftID)
```

Reply:
Yes, a mistake that will be corrected. Thanks!